### PR TITLE
refactor code

### PR DESCRIPTION
**Pull request by marcinkozakdev**

### CODE REVIEW

File: DieteticConsultationAPI.UnitTest/DieticianServiceTest.cs [hunk 1]
```diff
@@ -230,60 +229,24 @@ public async Task DeleteDietician_DieticianFound_ReturnNull()
...
-            // assert
-            await Assert.ThrowsAsync<NotFoundHttpException>(()=> _sut.DeleteDietician(dietician.Id));
-        }
-
         private Dietician SampleDietician()
         {
             return new Dietician()
             {
+                Id = 1,
                 FirstName = "Dominika",
```
Comment (MathsWho) [inline DieteticConsultationAPI.UnitTest/DieticianServiceTest.cs]: remove the real name and use fake one like "test1@test.com

File: DieteticConsultationAPI/Models/DietDto.cs [hunk 2]
```diff
@@ -1,15 +1,57 @@
...
+                Description = diet.Description,
+                CalorificValue = diet.CalorificValue,
+                ProhibitedProducts = diet.ProhibitedProducts,
+                RecommendedProducts = diet.RecommendedProducts,
+                Files = diet.Files?.Select(FileModelDto.For).ToList()
+            };
+
+        public static async ValueTask<DietDto?> BindAsync(HttpContext context)
+        {
+            var form = await context.Request.ReadFormAsync();
```
Comment (MathsWho) [inline DieteticConsultationAPI/Models/DietDto.cs:34]: Is this actually work ? Did you test it ?

File: DieteticConsultationAPI/Repositories/DietRepository.cs [hunk 3]
```diff
@@ -37,24 +37,25 @@ public async Task<Diet> AddOrUpdate(Diet diet)
...
-                .FirstOrDefaultAsync(d => d.Id == id);
-
-            _context.Diets.Remove(diet);
-            await _context.SaveChangesAsync();
+            if (await _context.Diets.FirstOrDefaultAsync(d => d.Id == id) is { } diet)
+            {
+                _context.Diets.Remove(diet);
+                await _context.SaveChangesAsync();
+            }
         }
```
Comment (MathsWho) [inline DieteticConsultationAPI/Repositories/DietRepository.cs:49]: Add here additional info when diet does not exist for a particular id, you can throw an exception to inform the end user of such purpose

File: DieteticConsultationAPI/Services/AccountService.cs [hunk 4]
```diff
@@ -46,13 +46,13 @@ public async Task<string> GenerateJwt(LoginDto dto)
             var user = await _accountRepository.Login(dto);
 
             if (user is null)
-                BadRequestHttpException.For("Invalid username or password");
+                IncorrectLogginException.For("Invalid username or password");
```
Comment (MathsWho) [inline DieteticConsultationAPI/Services/AccountService.cs]: You can pass userid as a parameter

**Review by MathsWho — APPROVED** [review]

File: DieteticConsultationAPI/Repositories/DietRepository.cs (hunk 3 above)
Comment (marcinkozakdev) [reply DieteticConsultationAPI/Repositories/DietRepository.cs:49]: added additional condition for deleted resources

**Pull request merged**